ldcbench.jsonld-gen/src/main/java/org/dice_research/ldcbench/jsonld/gen/JsonLDDataGenerator.java

```diff
@@ -0,0 +1,144 @@
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RDFDataMgr;
+import org.dice_research.ldcbench.ApiConstants;
+import org.dice_research.ldcbench.benchmark.DataGenerator;
+import org.dice_research.ldcbench.graph.Graph;
+import org.dice_research.ldcbench.nodes.htmlembd.SimpleHEComponent;
+import org.dice_research.ldcbench.nodes.utils.TripleIterator;
+import org.dice_research.ldcbench.utils.tar.TarFileGenerator;
+import org.hobbit.core.rabbit.RabbitQueueFactory;
+import org.hobbit.core.rabbit.SimpleFileSender;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JsonLDDataGenerator extends DataGenerator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JsonLDDataGenerator.class);
+
+
+    protected static final String HTML_PREFIX = "<!DOCTYPE>\r\n<html>\r\n<head>\r\n   <title>Test</title>\r\n</head>\r\n<body>";
+    protected static final String HTML_SUFFIX = "</body>\n</html>";
+
+    protected static final String JSON_LD_PREFIX = "<script type=\"application/ld+json\">";
+    protected static final String JSON_LD_SUFFIX = "</script>\r\n";
+
+    protected void generateHtmlFiles(Graph[] graphs, File htmlOut) {
+
+
+        String htmlFileContent = new String();
+        int domainId = 1;
+
+        htmlFileContent += HTML_PREFIX;
+
+        TripleIterator iterator;
+        // convert Graph to jena.graph s.t. we can use the RDFDataMgr
+        Model model = ModelFactory.createDefaultModel();
+        org.apache.jena.graph.Graph modelGraph = model.getGraph();
+        int datasetId = 0;
+        for(Graph graph: graphs) {
+            for (int i = 0; i < graph.getNumberOfNodes(); ++i) {
+                iterator = new TripleIterator(graphs, domainId, resourceUriTemplates, accessUriTemplates, datasetId, i);
+                while(iterator.hasNext()) {
+                    modelGraph.add(iterator.next());
+                }
+            }
+            datasetId++;
+        }
+
+        OutputStream outputJLD = new OutputStream() {
+            private StringBuilder string = new StringBuilder();
+
+            @Override
+            public void write(int b) throws IOException {
+                this.string.append((char) b );
+            }
+
+            public String toString() {
+                return this.string.toString();
+            }
+        };
+        RDFDataMgr.write(outputJLD, model, Lang.JSONLD);
+
+        htmlFileContent += JSON_LD_PREFIX;
+        htmlFileContent += outputJLD.toString();
+        htmlFileContent += JSON_LD_SUFFIX;
+
+        htmlFileContent += HTML_SUFFIX;
+        System.out.println(htmlFileContent);
```

Comment:
Several things are a little bit strange in this method.

1. Everything is appended to the local variable `htmlFileContent` which is simply printed to the console. How does this help again to send the HTML file(s) to the crawler, later on? :confused: 
2. The JSON-LD is generated by Jena and transformed into bytes. After that, the bytes are transformed into characters again (by casting them so without any charset) and appended to a String. Finally, they should be printed to the given file, which means that they would be transformed into bytes again. I guess you see my point :wink: 

I would suggest ot open a (buffered) output stream to write to the file, write the HTML prefix and JSON-LD prefix, give the output stream to the jena library to write the JSON-LD model, and finally write the suffixes to the output stream. For transforming the Strings to bytes, a standard encoding like UTF-8 should be used.